src/main/java/org/folio/rest/impl/PoNumberAPI.java

```diff
@@ -4,50 +4,47 @@
 
 import javax.ws.rs.core.Response;
 
+import io.vertx.core.Vertx;
 import io.vertx.core.json.JsonObject;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.folio.rest.RestVerticle;
+import org.folio.dao.PostgresClientFactory;
+import org.folio.dao.order.OrderDAO;
 import org.folio.rest.jaxrs.model.PoNumber;
 import org.folio.rest.jaxrs.resource.OrdersStoragePoNumber;
 import org.folio.rest.persist.HelperUtils;
 import org.folio.rest.persist.PostgresClient;
-import org.folio.rest.tools.utils.TenantTool;
 import org.folio.rest.core.BaseApi;
 
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Context;
 import io.vertx.core.Handler;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public class PoNumberAPI extends BaseApi implements OrdersStoragePoNumber {
   private static final Logger log = LogManager.getLogger();
-  private static final String PO_NUMBER_QUERY = "SELECT nextval('po_number')";
+  private final PostgresClient pgClient;
+
+  @Autowired
+  private OrderDAO orderDAO;
+  @Autowired
+  private PostgresClientFactory pgClientFactory;
+
+  public PoNumberAPI(Vertx vertx, String tenantId) {
+    SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
+    pgClient = pgClientFactory.createInstance(tenantId);
+  }
 
   @Override
   public void getOrdersStoragePoNumber(Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    vertxContext.runOnContext((Void v) -> {
-      try {
-        String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
-
-        PostgresClient.getInstance(vertxContext.owner(), tenantId).select(PO_NUMBER_QUERY, ar -> {
-          try {
-            if (ar.succeeded()) {
-              String poNumber = ar.result().iterator().next().getLong(0).toString();
-              asyncResultHandler.handle(buildOkResponse(new PoNumber().withSequenceNumber(poNumber)));
-            } else {
-              log.error("Error with pg generating a new po number", ar.cause());
-              asyncResultHandler.handle(buildErrorResponse(ar.cause()));
-            }
-          } catch (Exception e) {
-            log.error("Error getting the generated po number", e);
-            asyncResultHandler.handle(buildErrorResponse(e));
-          }
-        });
-      } catch (Exception e) {
-        log.error("Error getting a new po number", e);
-        asyncResultHandler.handle(buildErrorResponse(e));
-      }
-    });
+    vertxContext.runOnContext((Void v) -> pgClient.withConn(conn -> orderDAO.getNextPoNumber(conn)
```

Comment:
wanted to check code locally but could not because it is from fork.
We had floating issues previously when returning respose by calling asyncResultHandler before DB conncetion closes.
So we need to move ')' to close connection before invoking onSuccess()

Reply:
fixed